lib/Textile/events.ts

```diff
@@ -147,8 +148,9 @@ class Events {
       } else {
         DeviceEventEmitter.removeListener(publicEvents[type as string], listener)
       }
+    } else {
+      throw new Error(`@textile/react-native-sdk: no event type: ${type}`)
     }
-    throw new Error(`@textile/react-native-sdk: no event type: ${type}`)
```

Comment:
For exactly this reason, I really try to avoid assuming the function is returned from inside a previous `if` clause like this. Too easy to not realize you're not returning early and then this line gets called no matter what. Better to be explicit and verbose.